rclpy/rclpy/action/client.py

```diff
@@ -229,7 +231,8 @@ def _remove_pending_result_request(self, future):
     # Start Waitable API
     def is_ready(self, wait_set):
         """Return True if one or more entities are ready in the wait set."""
-        ready_entities = self._client_handle.is_ready(wait_set)
+        with self._lock:
```

Comment:
I can definitely see that, and have adjusted appropriately

Reply:
all these variables are allocated in the stack memory, so even this is called in concurrent, there will be no racy condition for those variables.

https://github.com/ros2/rclpy/blob/43198cbfe11c84480b15478f76d5fa3e998e5039/rclpy/src/rclpy/action_client.cpp#L244-L270

because of above reason, `ActionServer` does not acquire the lock either.

https://github.com/ros2/rclpy/blob/43198cbfe11c84480b15478f76d5fa3e998e5039/rclpy/rclpy/action/server.py#L439-L447

besides that, `is_ready` will not be called concurrently even with `MultiThreadedExecutor` since that is called via `wait_for_ready_callbacks`.

Reply:
I've remove the locks in is_ready for client and server